app/src/main/java/com/asfoundation/wallet/repository/TrustPasswordStore.kt

```diff
@@ -0,0 +1,161 @@
+package com.asfoundation.wallet.repository
+
+import android.content.Context
+import android.os.Build
+import android.preference.PreferenceManager
+import android.widget.Toast
+import com.asfoundation.wallet.entity.ServiceErrorException
+import com.asfoundation.wallet.entity.Wallet
+import com.asfoundation.wallet.logging.Logger
+import com.asfoundation.wallet.util.KS
+import com.asfoundation.wallet.util.KS.ANDROID_KEY_STORE
+import com.wallet.pwd.trustapp.PasswordManager
+import io.reactivex.Completable
+import io.reactivex.Single
+import java.io.IOException
+import java.security.KeyStore
+import java.security.KeyStoreException
+import java.security.NoSuchAlgorithmException
+import java.security.SecureRandom
+import java.security.cert.CertificateException
+import java.util.*
+
+
+class TrustPasswordStore(private val context: Context,
+                         private val logger: Logger) :
+    PasswordStore {
+  companion object {
+    private val TAG = TrustPasswordStore::class.java.simpleName
+    private const val DEFAULT_WALLET = "0x123456789"
+  }
+
+  init {
+    migrate()
+  }
+
+  private fun migrate() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+      return
+    }
+    val pref =
+        PreferenceManager.getDefaultSharedPreferences(context)
+    val passwords = pref.all
+    for (key in passwords.keys) {
+      if (key.contains("-pwd")) {
+        val address = key.replace("-pwd", "")
+        try {
+          KS.put(context, address.toLowerCase(), PasswordManager.getPassword(address, context))
+        } catch (ex: Exception) {
+          Toast.makeText(context, "Could not process passwords.", Toast.LENGTH_LONG)
+              .show()
+          ex.printStackTrace()
+        }
+      }
+    }
+  }
+
+  override fun getPassword(
+      wallet: Wallet): Single<String> {
```

Comment:
Improve crappy kotlin codestyle to single line

Reply:
Done